### ISSUE TITLE
The first supplied array does not contain any binary object

### ISSUE THREAD

**Issue by Huiimin5** [HiLab-git/SSL4MIS#47]
Hi, thank you so much for your efforts in implementing all these methods.

After training on ACDC with labeled_num=3, the testing code on ACDC gives me this error "The first supplied array does not contain any binary object", which is raised near:
<img width="580" alt="image" src="https://user-images.githubusercontent.com/11154631/158099634-7365e755-2359-438d-a021-58a0b830f9a4.png">

The problem is that the trained model does not predict labeled 2 and label 3, which can be seen here:
<img width="488" alt="image" src="https://user-images.githubusercontent.com/11154631/158099750-3e922534-95fb-42cc-8f4b-9879339b645f.png">

The visualized dice on validation set shows that there is a performance peak at 600th step
<img width="496" alt="image" src="https://user-images.githubusercontent.com/11154631/158101622-d792dad3-c86a-4659-8a38-e63449fa0044.png">

The test performance on the validation set (at 600th step) looks normal
<img width="1394" alt="image" src="https://user-images.githubusercontent.com/11154631/158102270-22e899e3-98d6-43a9-b452-cb330d41a424.png">

It's confusing that on a test set image the same model does not prediction label 2 and 3 at all.  Have you met this issue and could you give me some advice?

Many thanks.

**Comment by Luoxd1996** [comment]
Hi,
Please use this function to measure the performance.

def calculate_metric_percase(pred, gt):
    pred[pred > 0] = 1
    gt[gt > 0] = 1
    if pred.sum() > 0 and gt.sum() > 0:
        dice = metric.binary.dc(pred, gt)
        hd95 = metric.binary.hd95(pred, gt, voxelspacing=[10, 1, 1])
        asd = metric.binary.asd(pred, gt, voxelspacing=[10, 1, 1])
        return dice, hd95, asd
    else:
        return 0, 50, 10.
Best,
Xiangde.

**Comment by Huiimin5** [comment]
Thanks a lot for your clarification.
By the way, in your work "Semi-Supervised Medical Image Segmentation via Cross Teaching between CNN and Transformer", you report standard deviation, is it calculated over each image? Basically for each foreground class, aside from taking the average over each image's dice, you also calculate the standard deviation, right?
Many thanks

**Comment by 2805111860** [comment]
测试时我将代码改成你上面发布的，但还是出现“The first supplied array does not contain any binary object”，请问大佬这个该怎么解决呢？感谢你的贡献。

**Comment by supgy** [comment]
> 测试时我将代码改成你上面发布的，但还是出现“The first supplied array does not contain any binary object”，请问大佬这个该怎么解决呢？感谢你的贡献。

have you soloved this problem?
